kaggle_environments/envs/halite/helpers.py

```diff
@@ -0,0 +1,780 @@
+from copy import deepcopy
+from enum import Enum, auto
+from typing import *
+
+# region Helper Classes and Methods
+Point = NewType('Point', Tuple[int, int])
+
+
+def translate_point(point: Point, offset: Point) -> Point:
+    """Returns (point.x + offset.x, point.y + offset.y)"""
+    (x1, y1) = point
+    (x2, y2) = offset
+    return x1 + x2, y1 + y2
+
+
+def wrap_point(point: Point, size: int) -> Point:
+    """
+    Returns (point.x % size, point.y % size)
+    If the point is not on the board of width and height size it will be wrapped around to fit on the board
+    """
+    (x, y) = point
+    return x % size, y % size
+
+
+def position_to_index(point: Point, size: int) -> int:
+    """
+    Converts a 2d position in the form (x, y) to an index in the observation.halite list.
+    See index_to_position for the inverse.
+    """
+    x, y = point
+    return (size - y - 1) * size + x
```

Comment:
This puts 0 in the lower left corner and (size - 1, size - 1) in the upper right which I personally found a lot more intuitive, curious to get others' thoughts on this though.

Reply:
Not sure I'm fully understanding, but my personal take is that `(0, 0)` as top left makes more sense because that's how 2d arrays work, and in particular the format used in the replay files.  I see how your way is intuitive relative to normal XY plane but IMO it's better to match the programming methodology.  But just a personal take and IIUC a lot / all of this is internal and the higher-level structure here means users aren't necessarily forced into either paradigm?

Reply:
@dster2 Yeah users would only even notice the coordinate system if they were constructing offsets by hand. If they only used NORTH/SOUTH/EAST/WEST they'd likely never even notice there are coordinates. I'll consider this a bit more, thanks for the feedback all.